hw/xbox/nv2a.c

```diff
@@ -1183,6 +1207,8 @@ typedef struct TextureKey {
     TextureShape state;
     uint64_t data_hash;
     uint8_t* texture_data;
+    uint64_t palette_hash;
```

Comment:
I considered that but XORing those 2 together could have horrendous results if any game set texture=palette (in case the palette is unused anything could happen).
Speaking of that, we only have to hash the palette if its actually used, however, we don't know if the palette is used at that point because the format doesn't store any information about it.
Maybe add a new flag to the format descriptors or a function like "bool format_has_palette(unsigned int format)" ?
Also I couldn't figure out how to properly update a hash value with the existing hash functions.